### PR TITLE
Compass fixes

### PR DESCRIPTION
**Pull request by dpgeorge**
This fixes the compass with the new DAL behaviour for calibration, and adds a new function.  Commits are:

- reference the "master" version of the DAL to get the relevant version
- get calibration working by giving the DAL control of the display during calibration
- add the get_field_strength() function [this seems to work very well, for instance gives a bigger reading when the microbit is close to a monitor]

I left the `is_calibrating()` function even though it always returns `False`.

### CODE REVIEW

**Comment by dpgeorge** [comment]
Addresses issue #166, if only temporarily.

**Comment by ntoll** [comment]
OK... that's a LOT more accurate..! :-)

Why leave in `is_calibrating`..? If it's not needed please remove it, squash the commits and then I'll merge.

**Comment by dpgeorge** [comment]
Updated.  I removed is_calibrating, and added help() for get_field_strength.

File: source/microbit/help.c [hunk 1]
```diff
@@ -93,11 +93,11 @@ STATIC const mp_doc_t help_table_instances[] = {
...
-    {&microbit_compass_is_calibrating_obj, "If micro:bit's compass is_calibrating() its accuracy, it sends True.\nIf it's not busy calibrating things, it sends False.\n"},
     {&microbit_compass_clear_calibration_obj, "Reset micro:bit's compass using clear_calibration() command.\nRun calibrate() to improve accuracy.\n"},
     {&microbit_compass_get_x_obj, "Return magnetic field detected along micro:bit's X axis.\nUsually, the compass returns the earth's magnetic field in micro-Tesla units.\nUnless...a strong magnet is nearby!\n"},
     {&microbit_compass_get_y_obj, "Return magnetic field detected along micro:bit's Y axis.\nUsually, the compass returns the earth's magnetic field in micro-Tesla units.\nUnless...a strong magnet is nearby!\n"},
     {&microbit_compass_get_z_obj, "Return magnetic field detected along micro:bit's Z axis.\nUsually, the compass returns the earth's magnetic field in micro-Tesla units.\nUnless...a strong magnet is nearby!\n"},
+    {&microbit_compass_get_field_strength_obj, "Return magnitude of magnetic field around micro:bit.\n"},
```
Comment (ntoll) [inline source/microbit/help.c]: How about s/magnitude/strength (or size). The help text is for a reading age of 11. ;-)

**Comment by ntoll** [comment]
This is great. Can you fix up the help text (sorry) before I merge it..?

File: source/microbit/help.c (hunk 1 above)
Comment (dpgeorge) [inline source/microbit/help.c]: Ok, will change.  I put magnitude because it's distinct from the directed values from get_{x,y,z}.

**Comment by dpgeorge** [comment]
Help text updated.

**Comment by ntoll** [comment]
Yay!

**Comment by dpgeorge** [comment]
Yay!  An SPI test is on the top of my list...

**Comment by ntoll** [comment]
Great stuff... it's the final item currently marked against a 1.0 release. ;-)

**Pull request merged**